cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/TimeScheduler.java

```diff
@@ -240,8 +267,7 @@ public void suspendSchedule(Id.Program program, SchedulableProgramType programTy
     throws NotFoundException, SchedulerException {
     checkInitialized();
     try {
-      scheduler.pauseTrigger(new TriggerKey(AbstractSchedulerService.scheduleIdFor(program, programType,
-                                                                                   scheduleName)));
+      scheduler.pauseTrigger(getGroupedTriggerKey(program, programType, scheduleName));
```

Comment:
Yes I have thought about this scenario while implementing. Here is my finding: 
- Trigger Paused state is mapped to our Suspended state.
- Now if suspend action is performed on a suspended trigger our handler in ProgramLifeCycleHttpHandler will send a conflict with schedule already suspended message": https://github.com/caskdata/cdap/blob/release/3.1/cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java#L434-434